lib/Xmds/Soap4.php

```diff
@@ -50,12 +50,17 @@ class Soap4 extends Soap
      * @param string $macAddress
      * @param null $xmrChannel
      * @param null $xmrPubKey
+     * @param string $osVersion
+     * @param string $osSdk
+     * @param string $manufacturer
+     * @param string $brand
+     * @param string $model
      * @return string
      * @throws GeneralException
      * @throws NotFoundException
      * @throws \SoapFault
      */
-    public function RegisterDisplay($serverKey, $hardwareKey, $displayName, $clientType, $clientVersion, $clientCode, $operatingSystem, $macAddress, $xmrChannel = null, $xmrPubKey = null)
+    public function RegisterDisplay($serverKey, $hardwareKey, $displayName, $clientType, $clientVersion, $clientCode, $operatingSystem, $macAddress, $xmrChannel = null, $xmrPubKey = null, $osVersion, $osSdk, $manufacturer, $brand, $model)
```

Comment:
Just a little hint while you're working on this still (I realise its a draft) - we're not adding additional parameters to this API, instead the details will appear in the existing `operatingSystem` parameter (either as JSON or plain string) as described in the issue.

Reply:
Hi, Dan! I really appreciate the review. Could you please take a look at the latest [commit ](https://github.com/xibosignage/xibo-cms/pull/2574/commits/48e7c9c0fab2c49afeb7846b0a06ac615c016a81) and let me know your thoughts? Thank you so much!